infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -138,26 +140,13 @@ public ManagementServer(ServerContext serverContext) {
             recovered = true;
         }
 
-        if ((Boolean) opts.get("--single")) {
-            String localAddress = opts.get("--address") + ":" + opts.get("<port>");
-
-            Layout singleLayout = new Layout(
-                    Collections.singletonList(localAddress),
-                    Collections.singletonList(localAddress),
-                    Collections.singletonList(new Layout.LayoutSegment(
-                            Layout.ReplicationMode.CHAIN_REPLICATION,
-                            0L,
-                            -1L,
-                            Collections.singletonList(
-                                    new Layout.LayoutStripe(
-                                            Collections.singletonList(localAddress)
-                                    )
-                            )
-                    )),
-                    0L
-            );
-
-            safeUpdateLayout(singleLayout);
+        synchronized (this.serverContext) {
```

Comment:
Is it possible to move this as a synchronized method into ServerContext. Two different objects are synchronizing on it. It will keep the synchronization in one place.